masu/external/kafka_msg_handler.py

```diff
@@ -194,23 +205,86 @@ def handle_message(msg):
 
 
     Args:
-        None
+        msg - Upload Service message containing usage payload information.
 
     Returns:
-        None
+        (String, dict) - String: Upload Service confirmation status
+                         dict: keys: value
+                               file: String,
+                               cluster_id: String,
+                               payload_date: DateTime,
+                               manifest_path: String,
+                               uuid: String,
+                               manifest_path: String
 
     """
     if msg.topic == HCCM_TOPIC:
         value = json.loads(msg.value.decode('utf-8'))
         try:
-            extract_payload(value['url'])
-            return SUCCESS_CONFIRM_STATUS
+            report_meta = extract_payload(value['url'])
+            return SUCCESS_CONFIRM_STATUS, report_meta
         except KafkaMsgHandlerError as error:
             LOG.error('Unable to extract payload. Error: %s', str(error))
-            return FAILURE_CONFIRM_STATUS
+            return FAILURE_CONFIRM_STATUS, None
     else:
         LOG.error('Unexpected Message')
-    return None
+    return None, None
+
+
+def get_account(provider_uuid):
+    """
+    Retrieve a provider's account configuration needed for processing.
+
+    Args:
+        provider_uuid (String): Provider unique identifier.
+
+    Returns:
+        (dict) - keys: value
+                 authentication: String,
+                 customer_name: String,
+                 billing_source: String,
+                 provider_type: String,
+                 schema_name: String,
+                 provider_uuid: String
+
+    """
+    all_accounts = []
+    try:
+        all_accounts = AccountsAccessor().get_accounts(provider_uuid)
+    except AccountsAccessorError as error:
+        LOG.info('Unable to get accounts. Error: %s', str(error))
+        return None
+
+    return all_accounts.pop()
+
+
+def process_report(report):
+    """
+    Process line item report and kick off summarization celery task.
+
+    Args:
+        report (Dict) - keys: value
+                        file: String,
+                        cluster_id: String,
+                        payload_date: DateTime,
+                        manifest_path: String,
+                        uuid: String,
+                        manifest_path: String
+    Returns:
+        None
+
+    """
+    cluster_id = report.get('cluster_id')
+    provider_uuid = utils.get_provider_uuid_from_cluster_id(cluster_id)
+    if provider_uuid:
+        account = get_account(provider_uuid)
+        LOG.info('Processing report for account %s', account)
+
+        reports_to_summarize = get_report_files(**account)
+        LOG.info('Processing complete for account %s', account)
+
+        async_id = summarize_reports.delay(reports_to_summarize)
+        LOG.info('Summarization celery uuid: %s', str(async_id))
```

Comment:
Inside this If-statement are three distinct log messages. Multiplied across every provider and every poll cycle, that could add up to a large logfile.

What do you think about doing one of these things?

1. Can we try to reduce this to one logging statement at the bottom of the if-statement.
2. If we really do need log messages for each step, let's look at how many of them can be DEBUG level instead of INFO.

Reply:
So just to clarify.  This code is not part of the polling path and it's not for every provider.  This code runs only when a Kafka message comes in with a report to process.

IMO these log statements are important to understand when each step of the processing happens.  Thinking primarily when triaging problems.  I'd prefer to leave this as is.